xapian-core/cluster/cluster.cc

```diff
@@ -304,6 +311,11 @@ Point::Point(const TermListGroup &tlg, const Document &document_)
 	double tf = 1 + log((double)wdf);
 	double idf = log(size / termfreq);
 	double wt = tf * idf;
+
+	// If weight of a term is zero, it doesn't contribute to the termlist
+	if (wt == 0)
+	    continue;
```

Comment:
`wdf >= 1`so `log(wdf) >= 0` so `tf >= 1`.

Hence to get `wt == 0` we need `idf == 0`, which only happens when `size == termfreq`.  Perhaps we should just check that above, and avoid two calls to `log()` (which is quite slow) in this case?

Is `termfreq > size` possible?  That would lead to negative `wt` which seems undesirable, so if it can happen we should probably exclude that too.